src/libs/kata-types/src/config/hypervisor/remote.rs

```diff
@@ -0,0 +1,85 @@
+use std::io::Result;
+use std::path::Path;
+use std::sync::Arc;
+
+use crate::{
+    config::{
+        default::{self, MAX_REMOTE_VCPUS, MIN_REMOTE_MEMORY_SIZE_MB},
+        ConfigPlugin,
+    },
+    eother, resolve_path,
+};
+
+use super::register_hypervisor_plugin;
+
+/// Hypervisor name for remote, used to index `TomlConfig::hypervisor`.
+pub const HYPERVISOR_NAME_REMOTE: &str = "remote";
+
+/// Configuration information for remote.
+#[derive(Default, Debug)]
+pub struct RemoteConfig {}
+
+impl RemoteConfig {
+    /// Create a new instance of `RemoteConfig`
+    pub fn new() -> Self {
+        RemoteConfig {}
+    }
+
+    /// Register the remote plugin.
+    pub fn register(self) {
+        let plugin = Arc::new(self);
+        register_hypervisor_plugin(HYPERVISOR_NAME_REMOTE, plugin);
+    }
+}
+
+impl ConfigPlugin for RemoteConfig {
+    fn name(&self) -> &str {
+        HYPERVISOR_NAME_REMOTE
+    }
+
+    /// Adjust the configuration information after loading from configuration file.
+    fn adjust_config(&self, conf: &mut crate::config::TomlConfig) -> Result<()> {
+        if let Some(remote) = conf.hypervisor.get_mut(HYPERVISOR_NAME_REMOTE) {
+            if remote.remote_hypervisor_socket.is_empty() {
+                remote.remote_hypervisor_socket =
+                    default::DEFAULT_REMOTE_HYPERVISOR_SOCKET.to_string();
+            }
+            resolve_path!(
+                remote.remote_hypervisor_socket,
+                "Remote hypervisor socket `{}` is invalid: {}"
+            )?;
+            if remote.remote_hypervisor_timeout == 0 {
+                remote.remote_hypervisor_timeout = default::DEFAULT_REMOTE_HYPERVISOR_TIMEOUT;
+            }
+            if remote.memory_info.default_memory == 0 {
+                remote.memory_info.default_memory = default::MIN_REMOTE_MEMORY_SIZE_MB;
+            }
+            if remote.memory_info.memory_slots == 0 {
+                remote.memory_info.memory_slots = default::DEFAULT_REMOTE_MEMORY_SLOTS
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Validate the configuration information.
+    fn validate(&self, conf: &crate::config::TomlConfig) -> Result<()> {
+        if let Some(remote) = conf.hypervisor.get(HYPERVISOR_NAME_REMOTE) {
+            if remote.remote_hypervisor_socket.is_empty() {
+                return Err(eother!("Remote hypervisor sock is not set"));
+            }
+            if remote.remote_hypervisor_timeout == 0 {
+                return Err(eother!("Remote hypervisor timeout is not set"));
+            }
```

Comment:
I believe the purpose of `validate()` is to perform a final validation of config's overall coherence after possible adjustments (like filling in defaults) have been made in `adjust_config()`.  I don't think it makes much sense to repeat here the checks that were already handled in `adjust_config()`.

Reply:
For instance, if remote hypervisor doesn't support memory hotplugging (as documented elsewhere in this PR) it could be a good idea to verify `default_maxmemory` is not set, or equals to `default_memory`.

A similar check should be performed for `default_vcpus`/`default_maxvcpus`, considering that vcpu hotplugging is not supported either.

We should also ensure that no nvdimm usage is configured, and ideally check for all other remote hypervisor limitations (unsupported settings, setting where just a subset of possible values are valid for the remote hypervisor etc.).